lighthouse-core/lib/traces/pwmetrics-events.js

```diff
@@ -18,6 +18,22 @@
 
 const log = require('../../../lighthouse-core/lib/log.js');
 
+/**
+ * @param {!Object} object
+ * @param {string} path
+ * @return {*}
+ */
+function safeGet(object, path) {
```

Comment:
> An alternative is to always to have the intermediate objects be created even when the audit fails.

I think it's reasonable not to have an `extendedInfo` if the audit failed, but if it didn't fail, all the `extendedInfo` paths are filled or explicitly null (or whatever) at the leaf properties.

I'm sighing via code review because I'm currently in the process of re-enabling closure compiler for the codebase and this will all have to be undone again :)

I don't want to block `plots/` on having to hunt down all the current failure modes though, so we can go with this for now and file an issue about `extendedInfo` structure guarantees (which having a type check test will help with :)